core/src/main/java/org/apache/druid/data/input/impl/TimedShutoffInputSourceReader.java

```diff
@@ -68,8 +68,11 @@ public TimedShutoffInputSourceReader(InputSourceReader delegate, DateTime shutof
   )
   {
     final Closer closer = Closer.create();
-    closer.register(delegateIterator);
+    // We must register the shutdow of the shutdown executor first. This will make sure that the shutdow of the
```

Comment:
nit: typo 'shutdow' -> 'shutdown'. 

Also, if you have to change things anyway, that is a lot of shutdown, how about replacing the last 2 sentences with something like "This will cause the executor to be closed last and ensure other resources which might depend on it can close successfully."

Reply:
Done